components/menu/menu-item-return.js

```diff
@@ -0,0 +1,119 @@
+import '../icons/icon.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { LocalizeStaticMixin } from '../../mixins/localize-static-mixin.js';
+import { MenuItemMixin } from './menu-item-mixin.js';
+import { menuItemStyles } from './menu-item-styles.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class MenuItemReturn extends RtlMixin(LocalizeStaticMixin(MenuItemMixin(LitElement))) {
+
+	static get styles() {
+		return [ menuItemStyles,
+			css`
+				:host {
+					display: flex;
+					padding: 0.75rem 1rem;
+				}
+
+				span {
+					flex: auto;
+					line-height: 1rem;
+					overflow-x: hidden;
+					overflow-y: hidden;
+					text-decoration: none;
+					text-overflow: ellipsis;
+					white-space: nowrap;
+				}
+
+				d2l-icon {
+					flex: none;
+					margin-right: 1rem;
+					margin-top: 0.1rem;
+				}
+
+				:host([dir="rtl"]) > d2l-icon {
+					margin-right: 0;
+					margin-left: 1rem;
+				}
+			`
+		];
+	}
+
+	static get resources() {
+		return {
+			'ar': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'en': {
+				'return': 'Returns to previous menu.',
+				'returnCurrentlyShowing': 'Returns to previous menu. You are viewing {menuTitle}.'
+			},
+			'es': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'fi': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'fr': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'ja': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'ko': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'pt': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'sv': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'tr': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'zh': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			},
+			'zh-tw': {
+				'return': '',
+				'returnCurrentlyShowing': '{menuTitle}'
+			}
+		};
+	}
+
+	firstUpdated() {
+		super.firstUpdated();
+		this.setAttribute('aria-label', this.localize('return'));
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldValue, propName) => {
+			if (propName === 'text') {
+				this.setAttribute('aria-label', this.localize('returnCurrentlyShowing', 'menuTitle', this.text));
```

Comment:
There have been a fix instances like this where we need to set an ARIA label/role on the host. This one is especially unfortunate since it's happening outside the render loop (since the host isn't part of rendering). The upcoming AOM stuff may help a bit -- especially with `role`, but maybe labels too. Not sure what else to do other that expose this as some form of computed property.

Reply:
Yeah it's not great. Would it be better to have some hidden span or use `d2l-offscreen` that has the `aria-label`?

Reply:
It needs to stay as-is for now, since the label needs to be on the same element as the `menuitem` role. And we can't move both of those down into a sub-`<div>` because then the host itself would need a role of `presentation` to indicate that it should be ignored. That might be worth investigating (although still has the issue with needing to apply a role), but we'd want to do it across all the item types -- I think we can leave that for now.

Reply:
The `d2l-menu` is responsible for managing the focus when moving between items as well.  I think we want the label on whatever gets focus.  If we try to move the focus deeper, that logic _could_ break.